smol/capp/generate/__init__.py

```diff
@@ -1,11 +1,13 @@
 """General tools for structure and occupancy generation."""
 
 from smol.capp.generate.enumerate import enumerate_supercell_matrices
+from smol.capp.generate.groundstate.upper_bound.solver import UpperBoundSolver
 from smol.capp.generate.random import generate_random_ordered_occupancy
 from smol.capp.generate.special.sqs import StochasticSQSGenerator
 
 __all__ = [
     "enumerate_supercell_matrices",
     "generate_random_ordered_occupancy",
     "StochasticSQSGenerator",
+    "UpperBoundSolver",
```

Comment:
I would suggest to keep the previous name "GroundStateSolver" and add the details on it being only the ground state for the given supercell and so an upperbound to the global ground state in the docstrings and documentation. I think calling it "UpperBoundSolver" will just confuse users and obfuscate the main purpose of the class

Reply:
We should leave space for implementing the lower-bound solver in the future. Probably, the correct way is to add extra explanation in docs.

Reply:
I think the current UpperBoundSolver docs is clear enough. Already edited the example notebook.